framework/CodeInterfaces/PHISICS/phisicsdata.py

```diff
@@ -0,0 +1,961 @@
+            summedDict[group][region][parameter] = sum
+    return summedDict
+      
+  def fileOutName(self,phisicsDataDict):
+    """
+      Puts in a list the INSTANT output file names based on the number of MPI.
+      The format is title.o-MPI
+      @ In, phisicsDataDict, dictionary, dictionary of variables passed by the interface
+      @ Out, instantOutputFileMPI, list, list of INSTANT output file names
+      @ Out, mrtauOutputFileMPI, list, list of MRTAU output file names
+    """
+    instantMPI = []
+    mrtauMPI = []
+    for mpi in xrange (0,phisicsDataDict['numberOfMPI']):
+      mrtauMPI.append(os.path.join(self.workingDir,phisicsDataDict['mrtauFileNameDict']['atoms_csv'].split('.')[0]+'-'+str(mpi)+'.csv')) # MRTAU files (numbers.csv)
+      if self.phisicsRelap is False:    # INSTANT files, no coupling with RELAP
+          instantMPI.append(phisicsDataDict['instantOutput']+'-'+str(mpi))
+      if self.phisicsRelap is True:     # INSTANT files, coupling with RELAP
+        if mpi == 0:
+          pass
+        else:
+          instantMPI.append('INSTout-'+str(mpi))
+    return instantMPI, mrtauMPI
+   
+  def cleanUp(self,jobTitle):
+    """
+      Removes the file that RAVEN reads for postprocessing.
+      @ In, jobTitle, string, job title name
+      @ Out, None
+    """
+    csvOutput = os.path.join(self.workingDir, jobTitle+'-'+self.perturbationNumber+'.csv')
+    if os.path.isfile(csvOutput):
+      os.remove(csvOutput)
+    with open (csvOutput,'wb'):
+      pass
+    
+  def getAbsoluteXS(self,xsOutput):
+    """
+      Parses the absolute cross section output generated by PHISICS.
+      @ In, xsOutput, string, filename of the XML file containing the absolute perturbed XS
+      @ Out, labelList, list of the XS labels
+      @ Out, valueList, list of the XS
+    """
+    labelList = []
+    valueList = []
+    tree = ET.parse(os.path.join(self.workingDir,xsOutput))
+    root = tree.getroot()
+    for materialXML in root.getiterator('library'):
+      for isotopeXML in materialXML.getiterator('isotope'):
+        reactionList = [j.tag for j in isotopeXML]
+        for k in xrange (0, len(reactionList)):
+          for groupXML in isotopeXML.getiterator(reactionList[k]):
+            individualGroup = [x.strip() for x in groupXML.attrib.get('g').split(',')]
+            individualGroupValues = [y.strip() for y in groupXML.text.split(',')]
+            for position in xrange(0,len(individualGroup)):
+              labelList.append(materialXML.attrib.get('lib_name')+'|'+isotopeXML.attrib.get('id')+'|'+reactionList[k]+'|'+individualGroup[position])
+              valueList.append(individualGroupValues[position])
+    return labelList, valueList
+    
+  def removeSpaces(self,line):
+    """
+      Removes the spaces.
+      @ In, line, string
+      @ Out, line, string, same line without the blank spaces
+    """
+    line = re.sub(r' ',r'',line)
+    return line
```

Comment:
?????? ```rstrip()```?????

Reply:
I can remove the methid if you want (I made one because I knew i was going to to that a few times). 

The method removes the spaces, not just \r or \n. So if I have a line: 
_**[space]**       TIME (days),U235 **[space]**   ,U238 **[space]** ,I135_
`self.removeSpaces(line)` turns it into _TIME(days),U235,U238,I135_
`line.rstrip()` turns it into _TIME (days),U235 **[space]**  ,U238 **[space]**  ,I135_
I do not want to have empty elements when I do `line.split(',')`